### PR TITLE
Add option to sync status LED state on split keyboards where both halves can be plugged into USB

### PR DESCRIPTION
**Pull request by MysteriousAges**
## Description

Some split keyboards, such as the keeb.io Sinc, can have the USB cable plugged into either half. When using the backlight capslock option, the state is currently correct only when the USB cable is plugged into the left half. If plugged into the right side, the dedicated capslock indication LED is always in its default state regardless of whether capslock is on or off. In the case of the Sinc, the LED is always on.

This PR attempts to rectify the issue by ensuring both halves of a split keyboard agree on the LED indicator state. In theory, this should allow keyboards where scroll lock, num lock, caps lock, etc. indicators on the slave half to correctly reflect the various states. I have gated this syncronization behind a `rules.mk` option to avoid injecting unneeded code into all split keyboards, or if people don't care about indication.

This feature, when compiled for the ATMega32U4 (the micro on the Sinc Rev. 1), adds an additional 48 bytes to the firmware. I have tested this on my Sinc Rev. 1 and am satisfied it works. I have not tested it on other keyboards, mainly because this is the only one I have access to. Since I have only added the flag to the `rules.mk` for the Sinc, no other keyboard will be impacted by this change at this time.

## Types of Changes

- [x ] Core
- [ ] Bugfix
- [ x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ x] Keymap/layout/userspace (addition or update)
- [x ] Documentation

## Checklist

- [ x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ x] My change requires a change to the documentation.
- [x ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
This looks to be a similar concept but different implementation of #11749

**Comment by MysteriousAges** [comment]
> This looks to be a similar concept but different implementation of #11749

And here I was thinking I was the only one that had this problem! Astonishingly few results when I was googling, now I see I was missing a few keywords that would've let me find your PR.

**Comment by drashna** [comment]
No worries!  It's come up on reddit and discord, and I've had the code sitting for a while before opening my PR.

Your implementation is less invasive, but my only concern is that the led function will be switching back and forth on the slave side due to the lack of usb status.   I could be wrong, and plan on testing that, as I'm not really happy with how invasive my code is.

File: common_features.mk [hunk 1]
```diff
@@ -503,6 +503,10 @@ ifeq ($(strip $(SPLIT_KEYBOARD)), yes)
     POST_CONFIG_H += $(QUANTUM_DIR)/split_common/post_config.h
     OPT_DEFS += -DSPLIT_KEYBOARD
 
+    ifeq ($(strip $(SYNC_STATUS_LEDS)), yes)
+        OPT_DEFS += -DSYNC_STATUS_LEDS
+    endif
+
```
Comment (drashna) [inline common_features.mk:509]: Would prefer that this was just a simple define, as we have with `split_mods_enable`

File: docs/feature_split_keyboard.md [hunk 2]
```diff
@@ -77,6 +77,12 @@ If you're using a custom transport (communication method), then you will also ne
 SPLIT_TRANSPORT = custom
 ```
 
+If your keyboard has USB ports on both halves and has indicator LEDs on one or both halves, add the following to your `rules.mk` to ensure state is correctly syncronized between halves.
+
+```make
+SYNC_STATUS_LEDS = yes
```
Comment (drashna) [inline docs/feature_split_keyboard.md:83]: same as above (see line 185/191)

**Comment by drashna** [comment]
The code here doesn't break anything, but it doesn't properly get set the led state. So the query functions don't work.  So for things like rgb matrix and oleds, the lock state would be wrong and not show up.

**Comment by tzarc** [comment]
Sorry about that, GitHub decided to delete the `develop` branch from upstream when we merged it, despite being told otherwise. Reopened.

**Pull request closed without merging**